module/data/activity/base-activity.mjs

```diff
@@ -0,0 +1,158 @@
+ * @property {string} consumption.scaling.max    Maximum number of scaling levels for this item.
+ * @property {object} duration
+ * @property {string} duration.value             Scalar value for the activity's duration.
+ * @property {string} duration.units             Units that are used for the duration.
+ * @property {string} duration.special           Description of any special duration details.
+ * @property {object} range
+ * @property {string} range.value                Scalar value for the activity's range.
+ * @property {string} range.units                Units that are used for the range.
+ * @property {string} range.special              Description of any special range details.
+ * @property {object} target
+ * @property {object} target.template
+ * @property {string} target.template.count      Number of templates created.
+ * @property {boolean} target.template.connected  Must all created areas be connected to one another?
+ * @property {string} target.template.type       Type of area of effect caused by this activity.
+ * @property {string} target.template.size       Size of the activity's area of effect on its primary axis.
+ * @property {string} target.template.width      Width of line area of effect.
+ * @property {string} target.template.height     Height of cylinder area of effect.
+ * @property {string} target.template.units      Units used to measure the area of effect sizes.
+ * @property {object} target.affects
+ * @property {string} target.affects.count       Number of individual targets that can be affected.
+ * @property {string} target.affects.type        Type of targets that can be affected (e.g. creatures, objects, spaces).
+ * @property {boolean} target.affects.choice     When targeting an area, can the user choose who it affects?
+ * @property {string} target.affects.special     Description of special targeting.
+ * @property {object} uses
+ * @property {number} uses.spent                 Number of uses that have been spent.
+ * @property {string} uses.max                   Formula for the maximum number of uses.
+ * @property {UsesRecoveryData[]} uses.recovery  Recovery profiles for this activity's uses.
+ */
+export default class BaseActivity extends foundry.abstract.DataModel {
+
+  /* -------------------------------------------- */
+  /*  Model Configuration                         */
+  /* -------------------------------------------- */
+
+  /** @override */
+  static LOCALIZATION_PREFIXES = ["DND5E.ACTIVITY"];
+
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  static defineSchema() {
+    return {
+      _id: new DocumentIdField({initial: () => foundry.utils.randomID()}),
+      type: new StringField({
+        blank: false,
+        required: true,
+        readOnly: true
+      }),
+      name: new StringField({initial: undefined}),
+      img: new FilePathField({initial: undefined, categories: ["IMAGE"]}),
+      system: new ObjectField(),
```

Comment:
Is this for module use? It should be possible for different activity types to simply extend the base schema, rather than having to use a polymorphic field like this. Maybe there are some things I have not considered. It might be helpful to have an example of a basic activity implementation (doesn't have to be a real one) so we can discuss.

Reply:
(Perhaps it was meant to be `flags`? That seems to be missing.)

Reply:
My plan was to design this in the same way actor or item documents are designed, where they have the base data which is shared by all and then type-specific data models inside `system`, which allows modules to easily add new types.